daemon/cmd/daemon.go

```diff
@@ -936,6 +934,13 @@ func NewDaemon(ctx context.Context, cleaner *daemonCleanup,
 		case option.Config.EgressMasqueradeInterfaces != "":
 			msg = fmt.Sprintf("BPF masquerade does not allow to specify devices via --%s (use --%s instead).",
 				option.EgressMasqueradeInterfaces, option.Devices)
+		case option.Config.TunnelingEnabled():
+			if !hasFullHostReachableServices() {
+				msg = "BPF masquerade requires full host reachable services enabled in tunnel mode."
+			}
+		case option.Config.EnableHostLegacyRouting:
+			msg = fmt.Sprintf("BPF masquerade requires host legacy routing (--%s=\"false\")",
+				option.EnableHostLegacyRouting)
```

Comment:
Why was this case added?

Reply:
this is when I set host legacy routing to true in my test, then non of the switch cases match, result in empty msg, then following `log.Warn(msg + " Falling back to iptables-based masquerading.")` logs empty msg + `Falling back to iptables-based masquerading`, user will be confused why falling back to iptables-based masquerading,  the `if !option.Config.EnableHostLegacyRouting {` at the end of the block check does not match, so the `log.Info` is missed too

Reply:
Are you sure this was related to legacy host routing? Could it not be because you had host reachable services disabled?

Reply:
after refactor and add `if msg`, does not make sense to add `option.Config.EnableHostLegacyRouting` switch case, the refactor seems not only improve the code, but also the logic, please check the new changes to see if it make sense